resilience4j-bulkhead/src/main/java/io/github/resilience4j/bulkhead/ThreadPoolBulkheadConfig.java

```diff
@@ -82,18 +91,23 @@ public int getMaxThreadPoolSize() {
         return maxThreadPoolSize;
     }
 
-    public int getCoreThreadPoolSize() {
-        return coreThreadPoolSize;
-    }
+    public int getCoreThreadPoolSize() { return coreThreadPoolSize; }
 
     public boolean isWritableStackTraceEnabled() {
         return writableStackTraceEnabled;
     }
 
-    public static class Builder {
+    public List<? extends ContextPropagator> getContextPropagator() {
+        return contextPropagators;
+    }
 
-        private ThreadPoolBulkheadConfig config;
+    public void setContextPropagators(List<? extends ContextPropagator> contextPropagators) {
```

Comment:
@RobWin , We have to add a setter so that Customizer can update config via bean. I only generated setter for contextPropagators.

Reply:
If you think this design is good than I can add `RegistryCustomizerXXX` in other part of R4J in a separate PR(so that we have better context in reviewing).

Reply:
I'm not sure. I've to think about it. All Configs are usually immutable by design.

Reply:
For `Customizer` interface to work we have to provide a way to mutate Configs.

Reply:
Yes, or the Customizer is applied before the ThreadPoolBulkheadConfig is even created. 
The customizers could be invoked in `ThreadPoolBulkheadConfigurationProperties.createThreadPoolBulkheadConfig`.

And a Customizer bean could look as follows: 

```
@Bean
public ThreadPoolPropertiesCustomizer customizer() {
        return (String instanceName, InstanceProperties properties) -> {
            if (instanceName.equals("bla")) {
                properties.setContextPropagator(TestThreadLocalContextPropagator.class);
            }
            return builder;
        };
}
```

Reply:
As an alternative we could add it in `ThreadPoolBulkheadConfiguration.threadPoolBulkheadRegistry`.
The method `bulkheadConfigurationProperties.createThreadPoolBulkheadConfig(name)` could return a builder instead of the final ThreadPoolBulkheadConfig. 
And the correct Customizer is applied to this builder.

Reply:
This could be done, I can try that!

Reply:
@RobWin , supporting Builders now. Please review!